doc/contributing.md

```diff
@@ -16,37 +16,217 @@ and we encourage all to read it carefully.
+* Participate in discussions at the quarterly PyGMT Community Meetings, which are
+  announced on the [forum governance page](https://forum.generic-mapping-tools.org/c/governance/9).
+* Spread the word about PyGMT or star the project!
 
+## Providing Feedback
 
-## Getting Help
+### Reporting a Bug
 
-Discussion often happens on GitHub issues and pull requests.
-In addition, there is a
+* Find the [*Issues*](https://github.com/GenericMappingTools/pygmt/issues) tab on the
+top of the GitHub repository and click *New Issue*.
+* Click on *Get started* next to *Bug report*.
+* **Please try to fill out the template with as much detail as you can**.
+* After submitting your bug report, try to answer any follow up questions about the bug
+  as best as you can.
+
+### Submitting a Feature Request
+
+* Find the [*Issues*](https://github.com/GenericMappingTools/pygmt/issues) tab on the
+  top of the GitHub repository and click *New Issue*.
+* Click on *Get started* next to *Feature request*.
+* **Please try to fill out the template with as much detail as you can**.
+* After submitting your feature request, try to answer any follow up questions as best
+  as you can.
+
+### Submitting General Comments/Questions
+
+There are several pages on the [Community Forum](https://forum.generic-mapping-tools.org/)
+where you can submit general comments and/or questions:
+
+* For questions about using PyGMT, select *New Topic* from the
+  [PyGMT Q&A Page](https://forum.generic-mapping-tools.org/c/questions/pygmt-q-a/11).
+* For general comments, select *New Topic* from the
+  [Lounge Page](https://forum.generic-mapping-tools.org/c/lounge/6).
+* To share your work, select *New Topic* from the
+  [Showcase Page](https://forum.generic-mapping-tools.org/c/Sow-your-nice-example-script/10).
+
+## General Guidelines
+
+### Resources for New Contributors
+
+Please take a look at these resources to learn about Git and pull requests (don't
+hesitate to [ask questions](#getting-help)):
+
+* [How to Contribute to Open Source](https://opensource.guide/how-to-contribute/).
+* [Git Workflow Tutorial](http://www.asmeurer.com/git-workflow/) by Aaron Meurer.
+* [How to Contribute to an Open Source Project on GitHub](https://egghead.io/courses/how-to-contribute-to-an-open-source-project-on-github).
+
+### Getting Help
+
+Discussion often happens on GitHub issues and pull requests. In addition, there is a
 [Discourse forum](https://forum.generic-mapping-tools.org/c/questions/pygmt-q-a) for
 the project where you can ask questions.
 
+### Setting up your Environment
+
+We highly recommend using [Anaconda](https://www.anaconda.com/download/) and the `conda`
+package manager to install and manage your Python packages.
+It will make your life a lot easier!
+
+The repository includes a conda environment file `environment.yml` with the
+specification for all development requirements to build and test the project.
```

Comment:
Move the below sentence to this paragraph since it seems they are related to each other?
```
See the [`environment.yml`](https://github.com/GenericMappingTools/pygmt/blob/master/environment.yml)
file for the list of dependencies and the environment name.
```

BTW, I also added `pygmt` after **environment name**.

```suggestion
specification for all development requirements to build and test the project.
See the [`environment.yml`](https://github.com/GenericMappingTools/pygmt/blob/master/environment.yml)
file for the list of dependencies and the environment name (`pygmt`).
```